a3p-integration/proposals/g:ymax1/test/ymax1.test.js

```diff
@@ -0,0 +1,345 @@
+
+  /** @type {BridgeAction} */
+  const invokeAction = {
+    // @ts-expect-error old type from npm
+    method: 'invokeEntry',
+    message: {
+      id,
+      targetName: 'ymaxControl',
+      method: 'getCreatorFacet',
+      args: [],
+      saveResult: { name: 'creatorFacet' },
+    },
+  };
+
+  await sendWalletAction(vsc, ymaxControlAddr, invokeAction);
+
+  t.deepEqual(await wup.invocation(id), {
+    name: 'creatorFacet',
+    passStyle: 'remotable',
+  });
+});
+
+test.serial('ymax told zoe that Access token is required', async t => {
+  const { [contractName]: instance } = fromEntries(
+    await vsc.readPublished(`agoricNames.instance`),
+  );
+
+  const id = makeActionId('open');
+
+  /** @type {BridgeAction} */
+  const redeemAction = {
+    method: 'executeOffer',
+    offer: {
+      id,
+      invitationSpec: {
+        source: 'contract',
+        // @ts-expect-error XXX Type 'import("...node_modules/@agoric/zoe/src/zoeService/types").Instance' is not assignable to type 'globalThis.Instance'.
+        instance,
+        publicInvitationMaker: 'makeOpenPortfolioInvitation',
+      },
+      proposal: {},
+    },
+  };
+
+  await sendWalletAction(vsc, ymaxControlAddr, redeemAction);
+
+  await t.throwsAsync(wup.offerResult(id), {
+    message: /missing properties \["Access"\]/,
+  });
+});
+
+test.serial('null upgrade existing instance with args override', async t => {
+  const id = makeActionId('upgrade');
+
+  /** @type {BridgeAction} */
+  const invokeAction = {
+    // @ts-expect-error old type from npm
+    method: 'invokeEntry',
+    message: {
+      id,
+      targetName: 'ymaxControl',
+      method: 'upgrade',
+      args: [
+        {
+          bundleId,
+          privateArgsOverrides,
+        },
+      ],
+    },
+  };
+
+  await sendWalletAction(vsc, ymaxControlAddr, invokeAction);
+
+  t.deepEqual(await wup.invocation(id), { passStyle: 'copyRecord' });
+
+  const { [contractName]: instance } = fromEntries(
+    await vsc.readPublished(`agoricNames.instance`),
+  );
+  t.is(
+    boardId(instance),
+    t.context.shared.instanceId,
+    `${contractName} has a the same instance`,
+  );
+
+  const vatInfo = await getVatInfoFromID(
+    /** @type {string} **/ (t.context.shared.vatID),
+  );
+
+  t.is(
+    /** @type {{incarnation: number}} */ (vatInfo.currentSpan()).incarnation,
+    1,
+  );
+
+  const { privateArgs } = JSON.parse(vatInfo.parameters().body.slice(1));
+  t.like(privateArgs, privateArgsOverrides);
+});
+
+test.serial('revoke contract control', async t => {
+  const id = makeActionId('revoke');
+
+  /** @type {BridgeAction} */
+  const invokeAction = {
+    // @ts-expect-error old type from npm
+    method: 'invokeEntry',
+    message: {
+      id,
+      targetName: 'ymaxControl',
+      method: 'revoke',
+      args: [],
+    },
+  };
+
+  await sendWalletAction(vsc, ymaxControlAddr, invokeAction);
+
+  t.deepEqual(await wup.invocation(id), { passStyle: 'undefined' });
+});
+
```

Comment:
I don't see a test for core eval upgrade after revoke.

Well, I suppose doing it via contractControl counts. hm.